connector/iterator/src/main/java/com/microsoft/accumulo/spark/processors/AvroRowMLeap.java

```diff
@@ -317,15 +317,25 @@ public Row next() {
             }
           }).asScala().toSeq());
 
-      DefaultLeapFrame resultDataFrame = this.transformer.transform(this.mleapDataFrame).get();
-      this.resultIterator = ((scala.collection.Iterable<Row>) resultDataFrame.collect()).iterator();
+      // This is faster, but leaks memory
```

Comment:
These changes look good. Out of curiosity, how much slower is this?

Reply:
10-20%. there seems to be significant over head associated with calling transform. maybe batch requests could help, but I'm not sure if it's worth the additional complexity.